plugins/modules/database/saphana/hana_query.py

```diff
@@ -0,0 +1,208 @@
+    host = params['host']
+    database = params['database']
+    encrypted = params['encrypted']
+
+    filepath = params['filepath']
+    query = params['query']
+
+    bin_path = f"/usr/sap/{sid}/HDB{instance}/exe/hdbsql"
+
+    if filepath is not None or query is not None:
+        present = True
+    else:
+        present = False
+
+    try:
+        command = [module.get_bin_path(bin_path, required=True)]
+    except Exception:
+        'hdbsql not found {0}. \r\nSTDOUT: {1}\r\n\r\nSTDERR: {2}'.format(
+            rc, out, err)
+
+    if query is not None:
+        if not isinstance(query, (str, list)):
+            module.fail_json(msg="the query option value must be a string or list, passed %s" % type(query))
+
+        for elem in query:
+            if not isinstance(elem, str):
+                module.fail_json(msg="the elements in query list must be strings, passed '%s' %s" % (elem, type(elem)))
+    if filepath is not None:
+        if not isinstance(filepath, (str, list)):
+            module.fail_json(msg="the query option value must be a string or list, passed %s" % type(filepath))
+        for elem in filepath:
+            if not isinstance(elem, str):
+                module.fail_json(msg="the elements in query list must be strings, passed '%s' %s" % (elem, type(elem)))
+    if present:
+        if encrypted is True:
+            command.extend(['-attemptencrypt'])
+        if autocommit is False:
+            command.extend(['-z'])
+        if host is not None:
+            command.extend(['-n', host])
+        if database is not None:
+            command.extend(['-d', database])
+        # -x Suppresses additional output, such as the number of selected rows in a result set.
+        command.extend(['-x', '-i', instance, '-u', user, '-p', password])
+
+        if filepath is not None:
+            command.extend(['-I'])
+            for p in filepath:
+                # makes a command like hdbsql -i 01 -u SYSTEM -p secret123# -I /tmp/HANA_CPU_UtilizationPerCore_2.00.020+.txt,
+                # iterates through files and append the output to var out.
+                command.extend([p])
+                (rc, out_raw, err) = module.run_command(command)
+                del command[-1]
+
+                out = out + csv_to_json(out_raw)
+        if query is not None:
+            for q in query:
+                command.extend([q])
+                # makes a command like hdbsql -i 01 -u SYSTEM -p secret123# "select user_name from users",
+                # iterates through multiple commands and append the output to var out.
+                (rc, out_raw, err) = module.run_command(command)
+                del command[-1]
+
+                out = out + csv_to_json(out_raw)
+        changed = True
+    else:
+        changed = False
+        out = "nothing to do, no command provided"
+
+    module.exit_json(changed=changed, message=rc, stdout=out,
+                     stderr=err, command=' '.join(command))
```

Comment:
This currently only returns the last executed command. How about returning a list of all executed commands?